src/lint/track_config.nim

```diff
@@ -621,6 +621,42 @@ proc satisfiesSecondPass(trackConfigContents: string; path: Path): bool =
   checkExercisesPCP(practiceExercises, result, path)
   checkExerciseSlugsAndForegone(exercises, result, path)
 
+proc getExerciseSlugs(data: JsonNode; k: string): HashSet[string] =
+  result = initHashSet[string]()
+  if data.kind == JObject:
+    if data.hasKey("exercises"):
+      let exercises = data["exercises"]
+      if exercises.kind == JObject:
+        if exercises.hasKey(k):
+          if exercises[k].kind == JArray:
+            for exercise in exercises[k]:
+              if exercise.kind == JObject:
+                if exercise.hasKey("slug"):
+                  let slug = exercise["slug"]
+                  if slug.kind == JString:
+                    let slugStr = slug.getStr()
+                    if slugStr.len > 0:
+                      result.incl slugStr
+
+proc checkExerciseDirsAreInTrackConfig(trackDir: Path; data: JsonNode;
+                                       b: var bool; path: Path) =
+  ## Sets `b` to `false` if there is an exercise directory that is
+  ## not an exercise `slug` in `data`.
+  for exerciseKind in ["concept", "practice"]:
+    let exerciseSlugs = getExerciseSlugs(data, exerciseKind)
+    if exerciseSlugs.len > 0:
+      let exercisesDir = trackDir / "exercises" / exerciseKind
+      if dirExists(exercisesDir):
+        for exerciseDir in getSortedSubdirs(exercisesDir):
+          let dirSlug = lastPathPart(exerciseDir.string)
+          if dirSlug notin exerciseSlugs:
+            let msg = &"{q $exercisesDir} contains a directory named {q dirSlug}, " &
+                      &"which is not a `slug` in the array of {exerciseKind} " &
+                       "exercises. Please add an entry for this exercise. If " &
```

Comment:
Maybe mention where the entry should be added (the config.json file)?

```suggestion
                       "exercises. Please add an entry for this exercise. If " &
```

Reply:
Tweaked it a bit. I didn't mention "the track-level `config.json` file" because our current error messages always print it at the end, like:

> \`./exercises/concept\` contains a directory named \`the-farm\`, which is not a \`slug\` in the array of concept exercises. Please add the exercise to that array. If the exercise is not ready to be shown on the website, please set its \`status\` value to "wip":
./config.json